src/control/cmd/daos_server/network.go

```diff
@@ -41,45 +44,51 @@ type networkCmd struct {
 type networkScanCmd struct {
 	cfgCmd
 	logCmd
-	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is the provider specified in daos_server.yml)"`
-	AllProviders   bool   `short:"a" long:"all" description:"Specify 'all' to see all devices on all providers.  Overrides --provider"`
+	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider or 'all' for all available (default is the provider specified in daos_server.yml)"`
 }
 
 func (cmd *networkScanCmd) Execute(args []string) error {
 	var provider string
 
 	if len(args) > 0 {
-		cmd.log.Debugf("An invalid argument was provided: %+v", args)
 		return errors.WithMessage(nil, "failed to execute the fabric and device scan.  An invalid argument was provided.")
 	}
 
-	switch {
-	case cmd.AllProviders:
-		cmd.log.Info("Scanning fabric for all providers")
-	case len(cmd.FabricProvider) > 0:
-		provider = cmd.FabricProvider
-		cmd.log.Infof("Scanning fabric for cmdline specified provider: %s", provider)
-	case len(cmd.config.Fabric.Provider) > 0:
+	provider = cmd.FabricProvider
```

Comment:
A clearer way of writing this might look like:
```go
provider := cmd.config.Fabric.Provider
switch {
case strings.EqualFold(cmd.FabricProvider, "all"):
        provider = ""
case cmd.FabricProvider != "":
        provider = cmd.FabricProvider
}
```
This way, you're starting with the assignment of the default value and only modifying it if one of the other criteria are met.

Reply:
Done.